mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -264,6 +264,9 @@ class KMeans @Since("1.5.0") (
   override def fit(dataset: Dataset[_]): KMeansModel = {
     val rdd = dataset.select(col($(featuresCol))).rdd.map { case Row(point: Vector) => point }
 
+    val instr = Instrumentation.create(this, rdd)
+    instr.logParams(featuresCol, predictionCol, k, initMode, initSteps, maxIter, seed, tol)
+
     val algo = new MLlibKMeans()
```

Comment:
one statistic that is usually very useful to get is the dimension of the vectors (`numFeatures`). One way to get it is to pass the instrumentation instance to `algo.run(rdd)` below, and mark this new method as `private[spark]`.